File: c_ds/train/basics/climbstair.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 1:
            return 1
        dp = [0 for _ in range(n+1)]
        dp[1] = 1
        dp[2] = 2
        for i in range(3, n+1):
            dp[i] = dp[i-1] + dp[i-2]
        return dp[n]


def main():
    s = Solution()
    print(s.climbStairs(1))
    print(s.climbStairs(2))
    print(s.climbStairs(6))


if __name__ == "__main__":
    main()

File: c_ds/train/basics/climbstair.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1) return 1;
        vector<int> steps(n+1, 0);
        steps[1] = 1;
        steps[2] = 2;
        for (auto level = 3; level < n+1; level++){
            steps[level] = steps[level - 1] + steps[level - 2];
        }
        return steps[n];
    }
};


int main(){
    Solution s;
    cout << s.climbStairs(1) << endl;
    cout << s.climbStairs(2) << endl;
    cout << s.climbStairs(4) << endl;
}